test/tools/jsontests/StateTests.cpp

```diff
@@ -49,7 +49,7 @@ void doStateTests(json_spirit::mValue& _v, bool _fillin)
 		string testname = i.first;
 		json_spirit::mObject& o = i.second.get_obj();
 
-		if (_fillin)
+		if (_fillin && !Options::get().createRandomTest)
```

Comment:
How many combinations are possible here?
* [ ] _fillin && !createRandom
* [ ] _fillin && createRandom
* [ ] !_fillin && !createRandom
* [ ] !_fillin && createRandom

We should try to remove unnecessary combinations or throw errors on meaningless combinations.